src/OrchardCore.Modules/OrchardCore.ContentFields/Handlers/ContentPickerContentHandler.cs

```diff
@@ -0,0 +1,53 @@
+using System.Collections.Generic;
+using System.Globalization;
+using System.Threading.Tasks;
+using Fluid;
+using Fluid.Values;
+using OrchardCore.ContentFields.Settings;
+using OrchardCore.ContentManagement;
+using OrchardCore.ContentManagement.Handlers;
+using OrchardCore.ContentManagement.Models;
+using OrchardCore.Liquid;
+using OrchardCore.Localization;
+
+namespace OrchardCore.ContentFields.Handlers
+{
+    public class ContentPickerContentHandler : ContentHandlerBase
+    {
+        private readonly ILiquidTemplateManager _templateManager;
+
+        public ContentPickerContentHandler(ILiquidTemplateManager templateManager)
+        {
+            _templateManager = templateManager;
+        }
+
+        public async override Task GetContentItemAspectAsync(ContentItemAspectContext context)
+        {
+            await context.ForAsync<ContentPickerAspect>(async aspect =>
+            {
+                var settings = aspect.ContentPickerFieldSettings as ContentPickerFieldSettings;
+
+                if (settings != null) {
+                    aspect.Title = await GetContentPickerItemExtendedDescription(context.ContentItem, aspect.Culture, settings.TitlePattern, context.ContentItem.ToString());
+                    aspect.Description = await GetContentPickerItemExtendedDescription(context.ContentItem, aspect.Culture, settings.DescriptionPattern, string.Empty);
+                }
+            });
+        }
+
+        private async Task<string> GetContentPickerItemExtendedDescription(ContentItem contentItem, CultureInfo culture, string pattern, string defaultValue)
+        {
+            var description = defaultValue;
```

Comment:
```suggestion
            var result = defaultValue;
```

Reply:
Here, I don't think you need to do this. This adds string allocation for nothing. You could also use defaultValue and morph its value and return it.